sbtgen/src/main/scala/izumi/sbtgen/Renderer.scala

```diff
@@ -86,6 +86,26 @@ class Renderer(
   protected val index: Map[ArtifactId, Artifact] = makeIndex(aggregates)
   protected val settingsCache = new mutable.LinkedHashMap[String, Int]()
 
+  protected val configuredGroups: Set[String] = {
+    def flattenGroups(groups: Set[Group]): Set[Group] = {
+      groups.foldLeft(Set.empty: Set[Group])((agg, g) => agg ++ unpackGroup(g))
+    }
+
+    def unpackGroup(group: Group): Set[Group] = {
+      if (group.deps.isEmpty) Set(Group(group.name))
+      else Set(Group(group.name)) ++ group.deps.foldLeft(Set.empty: Set[Group])((agg, g) => agg ++ unpackGroup(g))
+    }
+
+    val allGroups = for {
+      agg <- aggregates
```

Comment:
```suggestion
      aggregates.map(_.groups) ++ artifact.map(_.groups)
```
Note: in general you might want to avoid for-comprehensions for collections and reserve them for monads with a single return (effects, Option/Try/Either)